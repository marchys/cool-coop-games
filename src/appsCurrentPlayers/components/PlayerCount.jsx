import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'react-emotion';

const Container = styled('div')`
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: green;
`;

export default function PlayerCount({ appId, playerCount }) {
  return (
    <Container>
      <p
        className={css`
          color: mediumspringgreen;
        `}
      >
        ID: {appId}
      </p>
      <p className={css({ color: 'cornsilk' })}>Count: {playerCount}</p>
    </Container>
  );
}

PlayerCount.propTypes = {
  appId: PropTypes.string.isRequired,
  playerCount: PropTypes.number.isRequired,
};
