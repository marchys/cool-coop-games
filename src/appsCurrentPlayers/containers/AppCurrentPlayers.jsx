import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import compose from 'lodash/fp/compose';

import Loading from 'components/Loading';
import actions from 'appsCurrentPlayers/redux/actions';
import selectors from 'appsCurrentPlayers/redux/selectors';
import propTypes from 'appsCurrentPlayers/redux/propTypes';

export class AppCurrentPlayers extends React.Component {
  componentDidMount() {
    const { appId, getCurrentPlayers } = this.props;
    getCurrentPlayers(appId);
  }

  render() {
    const { appId, currentPlayers } = this.props;

    if (!currentPlayers || !currentPlayers.loaded) {
      return <Loading />;
    }

    return (
      <div>
        <p>{appId}</p>
        <p>{currentPlayers.playerCount}</p>
      </div>
    );
  }
}

AppCurrentPlayers.propTypes = {
  getCurrentPlayers: PropTypes.func.isRequired,
  appId: PropTypes.string.isRequired,
  currentPlayers: propTypes.currentPlayers,
};

AppCurrentPlayers.defaultProps = {
  currentPlayers: void 0,
};

const mapStateToProps = (state, props) =>
  createStructuredSelector({
    currentPlayers: selectors.currentPlayersByAppId(props.appId),
  })(state, props);

const connectToStore = connect(
  mapStateToProps,
  {
    getCurrentPlayers: actions.getCurrentPlayers,
  },
);

export default compose(connectToStore)(AppCurrentPlayers);
