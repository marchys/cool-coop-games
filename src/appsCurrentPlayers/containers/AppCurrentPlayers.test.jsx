import React from 'react';
import { shallow } from 'enzyme';

import Loading from 'components/Loading';
import PlayerCount from '../components/PlayerCount';
import { AppCurrentPlayers } from './AppCurrentPlayers';

function setup({ appId = '74933', getCurrentPlayers = jest.fn(), ...props } = {}) {
  const wrapper = shallow(
    <AppCurrentPlayers appId={appId} getCurrentPlayers={getCurrentPlayers} {...props} />,
  );

  return {
    wrapper,
    get loading() {
      return wrapper.find(Loading);
    },
    get playerCount() {
      return wrapper.find(PlayerCount);
    },
  };
}

it('should render', () => {
  const { wrapper } = setup();
  expect(wrapper.getElement()).not.toBe(null);
});

test('should show loading if currentPlayers is undefined', () => {
  const wrapper = setup({ currentPlayers: void 0 });

  expect(wrapper.loading).toExist();
});

test('should show loading if currentPlayers is not loaded', () => {
  const wrapper = setup({ currentPlayers: { id: '74933', loading: true, loaded: false } });

  expect(wrapper.loading).toExist();
});

test('should render PlayerCount if loaded', () => {
  const currentPlayers = { id: '74933', loading: false, loaded: true, playerCount: 2000 };
  const appsCurrentPlayers = setup({ currentPlayers });

  expect(appsCurrentPlayers.playerCount).toMatchSnapshot();
});
