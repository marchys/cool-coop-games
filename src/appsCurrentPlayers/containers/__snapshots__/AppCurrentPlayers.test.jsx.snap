// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayerCount if loaded 1`] = `
<PlayerCount
  appId="74933"
  playerCount={2000}
/>
`;
